<launch>

  # 옵션(단위: m, rad, s)
  <arg name="share" default="true"/>                  # 공유제어 실행여부

  <arg name="ang_vel_intersection" default="0.2"/>    # 교차로에서의 회전속도
  <arg name="ang_fwd_threshold" default="0.6"/>       # 직진경로인정 최소각도
  <arg name="dist_alarm_1" default="5.0"/>            # 알림거리 1
  <arg name="dist_alarm_2" default="3.0"/>            # 알림거리 2: 명령 요청지점
  <arg name="lin_vel_joy" default="0.69"/>            # 조이스틱 최고 선속도 0.69
  <arg name="ang_vel_joy" default="1.50"/>            # 조이스틱 최고 각속도 3.67
  
  # 이동로봇 제어
  <node pkg="shared_control" type="task_plan.py" name="task_planner" output="screen" if="$(arg share)">
    <param name="ang_vel_intersection" value="$(arg ang_vel_intersection)"/>
    <param name="ang_fwd_threshold" value="$(arg ang_fwd_threshold)"/>
    <param name="dist_alarm_1" value="$(arg dist_alarm_1)"/>
    <param name="dist_alarm_2" value="$(arg dist_alarm_2)"/>
  </node>

  # 인터페이스
  <node pkg="joy" type="joy_node" name="joy_node"/>
  <node pkg="shared_control" type="interface.py" name="interfacer" output="screen">
    <param name="lin_vel_joy" value="$(arg lin_vel_joy)"/>
    <param name="ang_vel_joy" value="$(arg ang_vel_joy)"/>
    <param name="camera" value="camera/color/image"/>
  </node>
  <node pkg="rviz" type="rviz" name="rviz" respawn="true" args="-d $(find shared_control)/rviz/start.rviz"/>

</launch>
